projects/core/src/occ/adapters/product/converters/occ-product-search-page-normalizer.ts

```diff
@@ -20,11 +29,38 @@ export class OccProductSearchPageNormalizer
       ...target,
       ...(source as any),
     };
+    this.normalizeFacetValues(target);
     if (source.products) {
       target.products = source.products.map(product =>
         this.converterService.convert(product, PRODUCT_NORMALIZER)
       );
     }
     return target;
   }
+
+  /**
+   *
+   * In case there are so-called `topValues` given for the facet values,
+   * we replace the facet values by the topValues, simple because the
+   * values are obsolete.
+   *
+   * `topValues` is a feature in the adaptive search which can limit a large
+   * amount of facet values to a small set (5 by default). As long as the backend
+   * provides all facet values AND topValues, we normalize the data to not bother
+   * the UI with this specific feature.
+   */
+  private normalizeFacetValues(target: ProductSearchPage): void {
+    if (target.facets) {
+      target.facets.map((facet: Facet) => {
+        if ((<any>facet).topValues) {
+          facet.topValueCount = (<any>facet).topValues.length;
+          // no need to keep the array of topValues, as they duplicate the data.
+          delete (<any>facet).topValues;
```

Comment:
I'm thinking that it might be a better idea to avoid using `delete` but instead implement this method to be immutable and return a new object.
Would something like this work?
```ts
return {
...target,
facet: {
...target.facet,
topValues: undefined
}
}
```

Reply:
Agree, thx for the input, i'll have a 2nd look.